src/components/program-progress/ProgramProgressCourses.jsx

```diff
@@ -60,7 +107,10 @@ const ProgramProgressCourses = ({ courseData }) => {
                     {course.isEnded ? 'View Archived Course' : 'View Course'}
                   </a>
                   {course.certificateUrl
-                && renderCertificatePurchased()}
+                    ? renderCertificatePurchased()
+                    : course.upgradeUrl && !(course.expired === true)
+                    && getCertificatePriceString(course)
+                    && renderCertificatePriceMessage(course)}
```

Comment:
Conditional part after `:` is hard to understand, can we move this to a function?

Reply:
I've done changes. Please take a look.